src/criteria_assessment/criteria.jl

```diff
@@ -148,6 +151,47 @@ function setup_region_routes(config, auth)
         return file(mask_path; headers=COG_HEADERS)
     end
 
+    @get auth("/suitability/assess/{reg}/{rtype}") function (
+        req::Request, reg::String, rtype::String
+    )
+        # somewhere:8000/suitability/assess/region-name/reeftype?criteria_names=Depth,Slope&lb=-9.0,0.0&ub=-2.0,40
+        # 127.0.0.1:8000/suitability/assess/Cairns-Cooktown/slopes?Depth=-4.0:-2.0&Slope=0.0:40.0&Rugosity=0.0:6.0
+
+        qp = queryparams(req)
+        @debug "In region assessment route"
+        return assess_region(reg_assess_data, reg, qp, rtype, config)
+    end
+
+    @get auth("/suitability/site-suitability/{reg}/{rtype}") function (
+        req::Request, reg::String, rtype::String
+    )
+        # 127.0.0.1:8000/suitability/site-suitability/Cairns-Cooktown/slopes?Depth=-4.0:-2.0&Slope=0.0:40.0&Rugosity=0.0:6.0&SuitabilityThreshold=95&xdist=450&ydist=50
+
+        qp = queryparams(req)
+        criteria_names = [
+            "Depth",
+            "Benthic",
+            "Geomorphic",
+            "Slope",
+            "Turbidity",
+            "WavesHs",
+            "WavesTp",
+            "Rugosity",
+            "PortDistSlopes",
+            "PortDistFlats"
+        ]
+        criteria_qp = filter(k -> string(k.first) ∈ criteria_names, qp)
+
+        assessment_qp = filter(
+            k -> string(k.first) ∈ ["SuitabilityThreshold", "xdist", "ydist"], qp
+        )
```

Comment:
These are creating an anonymous function (`k ->`) on every call to the endpoint/function (triggering compilation and eventually the GC).

Can you explain what this is doing and I'll help think of a better approach

Reply:
It was to filter the dictionary that queryparams(request) produces so that I can have one dictionary with the user's environmental criteria and one dictionary with the user's site assessment criteria (box, suitability threshold etc). But maybe it's pointless anyway since they are both passed on to the same function

Reply:
> These are creating an anonymous function (`k ->`) on every call to the endpoint/function (triggering compilation and eventually the GC).

Are you sure this is true?

Found this very old issue https://github.com/JuliaLang/julia/issues/1864 fixed by https://github.com/JuliaLang/julia/pull/13412 in 2016
https://stackoverflow.com/questions/57740735/are-fast-anonymous-functions-automatic-now
https://github.com/timholy/FastAnonymous.jl claims Julia 0.5 made it obsolete.

Benchmark tools doesn't report difference 
```julia
using BenchmarkTools

list = ["ab","ac", "ba", "bb", "ca", "cb"]

function anon_fn(data)
    filter(x -> startswith(x, "a"), data)
end

function anon_fn2(data, prefix::String)
    filter(x -> startswith(x, prefix), data)
end

function startswith_a(data)
    startswith(data, "a")
end

function defined_fn(data)
    filter(startswith_a, data)
end


@benchmark anon_fn(data) setup=(data=list)
@benchmark anon_fn2(data, "a") setup=(data=list)

@benchmark defined_fn(data) setup=(data=list)
```

The anonymous function is slightly faster?
![image](https://github.com/user-attachments/assets/f549402c-3d1f-4093-9699-6d3fa2383f17)

Reply:
Last I checked it was true... How are you invoking the `@btime`?

Also, check out Chairmarks.jl

Reply:
Sorry, didn't see the code snippets before for some reason. I'll take a look shortly